apps/web/src/components/ai/ChatProvider.tsx

```
@@ -86,6 +101,13 @@ export function ChatProvider({ children }: { children: React.ReactNode }) {
     const selectedProviderRef = useRef(selectedProvider);
     selectedProviderRef.current = selectedProvider;
 
+    // Which model the composer is actually pinned to right now. A thread that
+    // exists owns its pin; before that, the browser-level seed does. Derived
+    // rather than stored so the two can never disagree.
+    const selectedModel = conversationId ? threadModel : seedModel || null;
+    const selectedModelRef = useRef(selectedModel);
+    selectedModelRef.current = selectedModel;
```

ESLint

[error] Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

/home/jailuser/git/apps/web/src/components/ai/ChatProvider.tsx:109:5
  107 |     const selectedModel = conversationId ? threadModel : seedModel || null;
  108 |     const selectedModelRef = useRef(selectedModel);
> 109 |     selectedModelRef.current = selectedModel;
      |     ^^^^^^^^^^^^^^^^^^^^^^^^ Cannot update ref during render
  110 |
  111 |     const chat = useChat({ id: 'ever-works-chat', transport });
  112 |

(react-hooks/refs)